### PR TITLE
Add test that shards, exports and runs with IREE a Unet Resnet block

### PR DESCRIPTION
**Pull request by sogartar**
The test targets llvm-cpu with the local-task IREE driver.

### CODE REVIEW

**Comment by sogartar** [comment]
This PR depends on
https://github.com/iree-org/iree-turbine/pull/73
https://github.com/iree-org/iree-turbine/pull/20

File: sharktank/tests/models/punet/sharded_resnet_block_with_iree_test.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    assert sharded_input_image.shard_count == sharded_input_time_emb.shard_count
+    shard_count = sharded_input_image.shard_count
+    logger.info(f"Running module.")
+    hal_driver = iree.runtime.get_driver("local-task")
+    vm_instance = iree.runtime.VmInstance()
+    available_devices = hal_driver.query_available_devices()
+    # Use the same actual device for all devices.
+    devices = [
+        hal_driver.create_device(available_devices[0]) for _ in range(shard_count)
+    ]
```
Comment (sogartar) [inline sharktank/tests/models/punet/sharded_resnet_block_with_iree_test.py:61]: @stellaraccident, you mentioned that we can target multiple IREE CPU devices.
What I have done here is effectively equivalent to
```
iree-run-module \
  --device=local-task \
  --device=local-task \
  ....
```
This sets the assigns `local-task` to the 2 device promises in the model.
Is this what you had in mind?

**Comment by sogartar** [comment]
The test in this PR is failing with large numerical imprecisions compared to the expectation.

**Review by stellaraccident — APPROVED** [review]
Thanks. I may refactor/break this up into a better example at some point, but good to at least have the test for now.

**Comment by stellaraccident** [comment]
> The test in this PR is failing with large numerical imprecisions compared to the expectation.

If you can't find the imprecision quickly, get it submitted as a WIP with the precision check commented. Then will need to triage.

**Pull request merged**